### PR TITLE
Inventory Items API

### LINKED ISSUE #8: Inventory Items API
Add Inventory Items API to app.

### PR DESCRIPTION
**Pull request by michaelachrisco**
Fixed #8

### CODE REVIEW

File: app/controllers/api/v1/trucking/inventory_items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module Api
+  module V1
+    module Trucking
```
Comment (josephbridgwaterrowe) [inline app/controllers/api/v1/trucking/inventory_items_controller.rb]: Module should be Inventory, class should be ItemsController.

File: db/migrate/20150107233846_create_oauth_access_grants.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class CreateOauthAccessGrants < ActiveRecord::Migration
```
Comment (josephbridgwaterrowe) [inline db/migrate/20150107233846_create_oauth_access_grants.rb]: Where did these OAuth things come from, shouldn't they of been in master?

Comment (michaelachrisco) [inline db/migrate/20150107233846_create_oauth_access_grants.rb]: They were already in master. The unshema gem goes and makes the migrations easier by adding a more verbose set of schema migrations. These files were generated by the tool. 
https://github.com/neopoly/unschema

File: app/controllers/api/v1/trucking/inventory_items_controller.rb (hunk 1 above)
Comment (michaelachrisco) [inline app/controllers/api/v1/trucking/inventory_items_controller.rb]: So to be clear, the directory would then be: 
app/controllers/api/v1/inventory/items_controller.rb

Comment (josephbridgwaterrowe) [inline app/controllers/api/v1/trucking/inventory_items_controller.rb]: Yes.

File: db/migrate/20150107233846_create_oauth_access_grants.rb (hunk 2 above)
Comment (josephbridgwaterrowe) [inline db/migrate/20150107233846_create_oauth_access_grants.rb]: But it uses strings rather than symbols :sad:

Does it check for the existing migration?

Comment (michaelachrisco) [inline db/migrate/20150107233846_create_oauth_access_grants.rb]: No I think I am just going to get rid of the gem/migrations. Related ticket here: https://github.com/neopoly/unschema/issues/2

It doesn't really add anything to the current branch at the moment although I could see it being used later on.

**Comment by michaelachrisco** [comment]
@josephbridgwaterrowe I think we are ready to merge.

File: app/models/inventory_item_adapter.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    def connected_with_octopus?
+      ActiveRecord::Base.using(:grossman).connected_with_octopus?
+    end
+
+    def all
+      results = connection.execute(sanitized_sql(QUERY_ALL))
+      results.map(&:to_snake_keys)
+    end
+
+    def like_id_description(item_id, initem_description)
```
Comment (josephbridgwaterrowe) [inline app/models/inventory_item_adapter.rb]: How about like_id_and_description?

**Comment by josephbridgwaterrowe** [comment]
I don't see any UUID's being create and stored against items?

File: app/models/inventory_item_adapter.rb (hunk 3 above)
Comment (josephbridgwaterrowe) [inline app/models/inventory_item_adapter.rb]: I would also prefer to wait until we have implemented the current master into payroll.

**Comment by michaelachrisco** [comment]
The original Items we discussed were from grossman. We can add that functionality as a separate issue if you would like.

File: app/models/inventory_item_adapter.rb (hunk 3 above)
Comment (michaelachrisco) [inline app/models/inventory_item_adapter.rb]: Sounds good.

**Comment by josephbridgwaterrowe** [comment]
Sure.

**Comment by josephbridgwaterrowe** [comment]
I didn't mean to merge it, I pressed the wrong button... Oh well...

**Pull request merged**